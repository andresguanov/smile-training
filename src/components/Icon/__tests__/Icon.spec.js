import { shallowMount } from "@vue/test-utils"
import smIcon from "@components/Icon/Icon.vue"
let wrapper

describe("smIcon", () => {
  //#region Basic render
  it("renders default icon", () => {
    wrapper = shallowMount(smIcon, {
      propsData: {
        icon: "home"
      }
    })
    expect(wrapper.classes("is-medium")).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })
  //#endregion

  //#region Props
  it("sets width", () => {
    wrapper = shallowMount(smIcon, {
      propsData: {
        icon: "plus",
        width: "40px"
      }
    })
    expect(wrapper.attributes("style")).toBe("--color: #999999; width: 40px; height: 40px;")
  })

  it("sets height", () => {
    wrapper = shallowMount(smIcon, {
      propsData: {
        icon: "plus",
        height: "40px"
      }
    })
    expect(wrapper.attributes("style")).toBe("--color: #999999; width: 40px; height: 40px;")
  })
  it("should be a small icon", () => {
    wrapper = shallowMount(smIcon, {
      propsData: {
        icon: "plus",
        size: "small"
      }
    })
    expect(wrapper.classes()).toContain("is-small")
  })

  it("should be a large icon", () => {
    wrapper = shallowMount(smIcon, {
      propsData: {
        icon: "plus",
        size: "large"
      }
    })
    expect(wrapper.classes()).toContain("is-large")
  })
  //#endregion
})
