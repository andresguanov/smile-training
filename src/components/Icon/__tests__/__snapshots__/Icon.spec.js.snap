// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smIcon renders default icon 1`] = `
<svg class="icon is-medium" style="--color: #999999;">
  <use xlink:href="[object Object]#home"></use>
</svg>
`;
